pkg/network/domainspec/generators.go

```diff
@@ -253,3 +263,87 @@ func (b *MacvtapLibvirtSpecGenerator) discoverDomainIfaceSpec() (*api.Interface,
 		},
 	}, nil
 }
+
+type PasstLibvirtSpecGenerator struct {
+	vmiSpecIface *v1.Interface
+	domain       *api.Domain
+}
+
+func (b *PasstLibvirtSpecGenerator) Generate() error {
+	if _, err := os.Stat("/var/run/passt"); !os.IsNotExist(err) {
+		log.Log.Reason(err).Infof("passt already running")
+		return nil
+	}
+	_, err := exec.Command("mkdir", "/var/run/passt").Output()
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to create /var/run/passt dir")
+	}
+	_, err = exec.Command("touch", "/var/run/passt/passt.log").Output()
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to create /var/run/passt/passt.log")
+	}
+
+	ports := b.generatePorts()
+	// TODO 107 is the uid of qemu. Once we will have a proper packaging for passt 107 can be substituted with "qemu".
+	_, err = exec.Command("sh", "-c", fmt.Sprintf("/usr/bin/passt --runas 107 -e %s 2> /var/run/passt/passt.log", ports)).Output()
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to start passt")
+	}
+
+	// remove passt interface from domain spec devices interfaces
+	foundDomainInterface := false
+	ifaces := b.domain.Spec.Devices.Interfaces
+	for i, iface := range ifaces {
+		if iface.Alias.GetName() == b.vmiSpecIface.Name {
+			b.domain.Spec.Devices.Interfaces = append(ifaces[:i], ifaces[i+1:]...)
+			foundDomainInterface = true
+			break
+		}
+	}
```

Comment:
why is this being done?

Reply:
Same as done for slirp, the current solution bypass libvirt and interact directly with qemu.
Once the libvirt and qemu rpms will be ready the passt interface will be defined using hte regular way via libvirt.